types/react-native/index.d.ts

```diff
@@ -352,7 +352,7 @@ type TaskProvider = () => Task;
 type NodeHandle = number;
 
 // Similar to React.SyntheticEvent except for nativeEvent
-export interface NativeSyntheticEvent<T> extends React.BaseSyntheticEvent<T, NodeHandle, NodeHandle> {}
+export interface NativeSyntheticEvent<T> extends React.BaseSyntheticEvent<T, React.ElementRef<HostComponent<unknown>>, React.ElementRef<HostComponent<unknown>>> {}
```

Comment:
I guess this is for `nativeEvent`? Can you add some tests so we don't regress and can replay the diff in the future?

Reply:
So it is not for `nativeEvent` :) It is just for `event` that we receive in callbacks like `onChange`, `onPress` and so on. 
`BaseSynthetucEvent` is defined in this way: 
   ```
 interface BaseSyntheticEvent<E = object, C = any, T = any> {
        nativeEvent: E;
        currentTarget: C;
        target: T;
        bubbles: boolean;
        cancelable: boolean;
        defaultPrevented: boolean;
        eventPhase: number;
        isTrusted: boolean;
        preventDefault(): void;
        isDefaultPrevented(): boolean;
        stopPropagation(): void;
        isPropagationStopped(): boolean;
        persist(): void;
        timeStamp: number;
        type: string;
    }
```

So the parameters I changed are related to `target` and `currentTarget` properties.

Reply:
@eps1lon, are "tests" here just code examples ? Since I don't see any structure for that or run instructions.

Reply:
Gotcha, this is for `event` and the other change is for `event.nativeEvent`?!

>@eps1lon, are "tests" here just code examples ? 

Yes.

Reply:
Right so we have just two lines changed. The first one is for `event` and the second one is for `nativeEvent`. Both `event` and `nativeEvent` has `target` property and both are incorrect.

Reply:
By the way, @eps1lon , there is a related [discussion](https://github.com/DefinitelyTyped/DefinitelyTyped/discussions/62550) (just in case you haven't seen)

Reply:
@eps1lon , I`ve added small change for `LayoutEvent`. So we don't have `target` in it actually, but docs says that we should. Also real state of affair is in accordance with docs.
Moreover, I`ve added small "test" for event.target/event.nativeEvent.target.
What is left ? Is there anything that I should do to make this PR ready to merge ?